### ISSUE TITLE
Double footer rows cause double-thickness top border

### ISSUE THREAD

**Issue by czp13** [vaadin/flow-components#4424]
### Description

The outcome right now:
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/61667986/209308785-b0992eaa-5d14-4ccf-b652-219f02ac9c38.png">

Zoomed to the grid header:
<img width="1414" alt="image" src="https://user-images.githubusercontent.com/61667986/209308873-bfb842ab-a0f5-44dc-8111-70a27d0c6087.png">

### Expected outcome

It would be nicer to have a normal border, not this double, thick border.
Fixed it on the fly added style to the element:
<img width="1429" alt="image" src="https://user-images.githubusercontent.com/61667986/209308993-19ba079f-330a-4dea-a418-b3fce2b426dc.png">

Potential, but not nice fix:
```
vaadin-grid.css

:host [part~='header-cell'] {
border-top: 0!important;
}
```

### Minimal reproducible example

```
@Route("double-header-minimal-code-vs-4449")
public class DoubleHeaderMinimalCodeVS4449 extends VerticalLayout {
    public DoubleHeaderMinimalCodeVS4449() {
        setSizeFull();
        
        Grid<String> grid = new Grid<>();
        grid.setSizeFull();
        add(grid);

        Column<String> column = grid.addColumn(row -> row).setHeader("header");
        grid.appendFooterRow().getCell(column).setText("footer1");
        grid.appendFooterRow().getCell(column).setText("footer2");
    }
}
```

### Steps to reproduce

1. Copy the snippet
2. Check the double or thick top border on top of the grid.
3. Make sure we are removing the correct border, so not removing the topmost one but the one for the cell, otherwise it can cause another visual glitch like this (so the scrollbar does not have any border)
![image](https://user-images.githubusercontent.com/61667986/209135098-37a13280-5903-4bf0-a3b3-74a9c704c167.png)

### Environment
Vaadin version(s): 23.x (tested with latest 23.3.1, 23th of Dec)
OS: MacOs Monterey, M1 chipset

### Browsers
Issue is not browser related

**Comment by czp13** [comment]
Related to this JIRA ticket (but only came up as a side effect for another issue):
- https://support.vaadin.com/browse/VS-4449 

Also provided a workaround for the customer which shall work for now, but I feel not the nicest solution:
```
:host [part~='header-cell'] {
    /* Add your CSS styles here, for example: */
    /*background: #f5f;*/
    /*border-top: 0!important;*/
}

:host [id~='vaadin-grid-cell-1'] {
    /* Add your CSS styles here, for example: */
    /*background: #f5f;*/
    border-top: 0!important;
}
```

**Comment by jouni** [comment]
What’s the output of the minimal reproducible example? I assume that in the screenshots, the "right aligned" cell is a header cell. Is there another hidden header row perhaps?

**Comment by rolfsmeds** [comment]
Please provide an actual minimal reproducible example. We cannot reproduce that screenshot with the code provided.

**Comment by czp13** [comment]
Thanks, @jouni. Yes, correct! 
`Right aligned` is a header cell. 
 I have updated the minimum reproducible example with a better, more matching code.

Context:
I fetched this from a customer ticket, many screenshots were there, and many codes, and issues too. Next time I rather construct a new example, minimum code from scratch.

TLDR:
The problem is with the thick, double border-top in the grid.

**Comment by czp13** [comment]
@rolfsmeds @jouni I will fix this ticket today, bear with me. I just somehow lost my vaadin-23 problems repository, which was deleted today (intellij bug maybe), and I wanted to create this ticket ASAP, thought this example from the ticket works (as its worked for me, but then something is different).

Note:
- I did not apply any custom CSS, my CSS shared here was fixing the thick/double border.

**Comment by czp13** [comment]
@rolfsmeds @jouni 
Thank you for your patience. 
- I fixed my env, 
- recreated the issue with minimal code, 
- and added potential workaround fixes as well. 

This issue was reported by a customer, but does not have a warranty ticket on it right now.

**Comment by jouni** [comment]
My quick analysis of the issue: because there are two footer rows, a column group is created. This effectively creates a second header row as well, although it is empty and not visible. The Lumo theme hides the top border for the first header row cells, but it doesn't take into account if the cells are actually visible or not: https://github.com/vaadin/web-components/blob/master/packages/grid/theme/lumo/vaadin-grid-styles.js#L216-L219

It looks like there is a `hidden` attribute on the hidden header row, and we could use that the CSS to remove the top border of the first visible header row cells. Something like this:

```css
[part~='row'][hidden] + [part~='row'] [part~='header-cell'] {
  border-top: 0;
}
```
